### PR TITLE
Fixed #6133 added some docs for routes provider

### LINKED ISSUE #6133: [doc] Document RoutesProvider for use with Guice
### Play Version (2.5.x / etc)

2.5.x

### API (Scala / Java / Neither / Both)

Scala

### Expected Behavior

```
@Singleton
class RoutesProvider @Inject() (injector: Injector, environment: Environment, configuration: Configuration, httpConfig: HttpConfiguration) extends Provider[Router] {
  lazy val get = {
    val prefix = httpConfig.context

    val router = Router.load(environment, configuration).fold[Router](Router.empty)(injector.instanceOf(_))
    router.withPrefix(prefix)
  }
}
```

### PR DESCRIPTION
**Pull request by schmitch**
## Fixes

Fixes #6133

## Purpose

Adds more documentation about routing

### CODE REVIEW

File: documentation/manual/working/scalaGuide/advanced/routing/ScalaSirdRouter.md [hunk 1]
```diff
@@ -66,6 +66,14 @@ A router can be provided to the application as detailed in [[Application Entry p
 
 @[load](code/SirdAppLoader.scala)
 
+### Providing a DI router with Guice
+
+A SIRD router for applications which using Guice could be done with overriding the `GuiceApplicationLoader` and the `Provider[Router]`:
```
Comment (gmethvin) [inline documentation/manual/working/scalaGuide/advanced/routing/ScalaSirdRouter.md]: How about
> A SIRD router can be provided in Guice-based Play apps by overriding the `GuiceApplicationLoader` and the `Provider[Router]`:

File: documentation/manual/working/javaGuide/advanced/routing/code/GuiceAppLoader.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.Arrays;
+import java.util.List;
+
+//#load-guice1
+public class GuiceAppLoader extends GuiceApplicationLoader {
+
+    @Override
+    protected GuiceableModule[] overrides(ApplicationLoader.Context context) {
+        GuiceableModule[] modules = super.overrides(context);
+        GuiceableModule module = GuiceableModule$.MODULE$.fromPlayBinding(new BindingKey<>(play.api.routing.Router.class).toProvider(GuiceRouterProvider.class).eagerly());
```
Comment (gmethvin) [inline documentation/manual/working/javaGuide/advanced/routing/code/GuiceAppLoader.java]: We really should make this easier. It seems odd that we have to delegate to Scala code in this way. But I guess that's why we need the example...

File: documentation/manual/working/scalaGuide/advanced/routing/code/ScalaSimpleRouter.scala [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+}
+
+class ScalaGuiceAppLoader extends GuiceApplicationLoader {
+
+  protected override def overrides(context: ApplicationLoader.Context): Seq[GuiceableModule] = {
+    val v1 = GuiceApplicationLoader.defaultOverrides(context)
+    val v2: Seq[GuiceableModule] = Seq(bind[Router].toProvider[ScalaRoutesProvider])
+
+    v1 ++ v2
```
Comment (gmethvin) [inline documentation/manual/working/scalaGuide/advanced/routing/code/ScalaSimpleRouter.scala]: Probably isn't that much more clear than
```scala
protected override def overrides(context: ApplicationLoader.Context): Seq[GuiceableModule] = {
  super.overrides(context) :+ bind[Router].toProvider[ScalaRoutesProvider]
}
```

Comment (schmitch) [inline documentation/manual/working/scalaGuide/advanced/routing/code/ScalaSimpleRouter.scala]: actually that's what I wanted however `bind[Router].toProvider[ScalaRoutesProvider]` needs to be implicit converted and my IDE wouldn't allow me to convert it implicit without the explicit type, but maybe this is just a IDE issue I will check.

File: documentation/manual/working/javaGuide/advanced/routing/code/GuiceAppLoader.java (hunk 2 above)
Comment (schmitch) [inline documentation/manual/working/javaGuide/advanced/routing/code/GuiceAppLoader.java]: Yeah that's what I think, too. Especially that it is a Array makes it somewhat hard to override something

**Comment by schmitch** [comment]
@gmethvin I updated the PR to include your changes, however some things only worked out by forcing the implicit conversion in the overrides.
Also I have a question regarding the Java part, should I keep the Java version as is and create a wrapper in another PR or should I add a Wrapper directly in this version?

**Comment by gmethvin** [comment]
@schmitch I merged the PR as-is and backported. You can create another PR for the wrapper changes.

**Pull request merged**